django/db/models/fields/related.py

```diff
@@ -1314,7 +1314,7 @@ class ForeignObjectRel(object):
 
     # Reverse relations are always nullable (Django can't enforce that a
     # foreign key on the related model points to this model).
-    null = True
+    null = False
```

Comment:
Gotcha - entirely possible that the public `null=True` behavior is solely for M2M fields, yup. That's where I ran into it when upgrading REST framework to pass existing test suite against master.